src/Aptos/Signer.cpp

```diff
@@ -183,7 +225,19 @@ Proto::SigningOutput Signer::sign(const Proto::SigningInput& input) {
             throw std::runtime_error("Nft message payload not set");
         }
     };
-    auto payloadFunctor = [&input, &nftPayloadFunctor]() {
+    auto liquidStakingFunctor = [](const Proto::LiquidStaking& liquidStakingMessage) {
+        switch (liquidStakingMessage.liquid_stake_transaction_payload_case()) {
+        case Proto::LiquidStaking::kStake:
+            return tortugaStakePayload(liquidStakingMessage.smart_contract_address(), liquidStakingMessage.stake());
+        case Proto::LiquidStaking::kUnstake:
+            return tortugaUnStakePayload(liquidStakingMessage.smart_contract_address(), liquidStakingMessage.unstake());
+        case Proto::LiquidStaking::kClaim:
+            return tortugaClaimPayload(liquidStakingMessage.smart_contract_address(), liquidStakingMessage.claim());
+        case Proto::LiquidStaking::LIQUID_STAKE_TRANSACTION_PAYLOAD_NOT_SET:
+            throw std::runtime_error("Nft message payload not set");
```

Comment:
Can we return an empty transaction payload instead?

Reply:
Fixed